code-challenges.test.js

```diff
@@ -20,9 +20,35 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+describe("shuffledColors", () => {
+    it("remove the first item from the array and shuffle the remaining items", () => {
+    expect(shuffledColors(colors1)).toEqual(["yellow", "blue", "pink", "green"]);
+    expect(shuffledColors(colors2)).toEqual(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]);
+    })
+})
+// ReferenceError: array is not defined (This is a good error)
+
 // b) Create the function that makes the test pass.
 
+const shuffledColors = (newArray) => {
+    const array = newArray.shift(0);
+    for (let i = array.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        [array[i], array[j]] = [array[j], array[i]];
+    }
+    return array;
+}
 // Pseudo code:
+// input:Take in a array of colors
+// output: return an array with the first color missing and rest shuffled
+// i created a test for it and when i tested it i received a good error
+// i used the .shift  method to try and remove an item from the beginning
+// line 35 loops through the length of the array but skips the first index
+// math.floor returns a random floating-point then we use math.floor to returns an integer
+// This is where i got stuck, someone in my class mentioned to use the fisher yates method.
+// i searched it up and read upon it but i kept getting stuck receiving errors and couldnt figure it out.
+
+// Resource: https://www.geeksforgeeks.org/shuffle-a-given-array-using-fisher-yates-shuffle-algorithm/
```

Comment:
Oscar, you are so close.  The reason it's not working is because when you set array.shift() to a variable, you get the removed item.  So you're whole program is working with one color. You did well working through the tough spots . Great explanation in your pseudocode!